### PR TITLE
Feat: Tags Manager

### PR DESCRIPTION
**Pull request by chakflying**
# Description

- View tags and monitors belonging to that tag
- Edit tag name & color
- Delete a tag (option to delete monitors with that tag?)

## Type of change

- User interface (UI)
- New feature (non-breaking change which adds functionality)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

image

image

image

image

### CODE REVIEW

**Comment by karelkryda** [comment]
It looks amazing

**Comment by Authorises** [comment]
I'm thinking maybe instead of editing it there, there is just a list of the tags, with an edit button and a delete button and the edit button opens up a modal where you can edit it and see a list of monitors in the list and remove them, change the color using any hex color, and Maybe it would be cool if you could add an icon as well? and then you can save it there.

**Comment by chakflying** [comment]
There seems to be a lack of high quality color picker library for Vue 3 that supports dark mode. The one from [Vuetify](https://vuetifyjs.com/en/components/color-pickers/#usage) looks nice, but I'm not sure if we should pull in something that big. We could maintain our own, but I'm not sure if it's worth the trouble.

**Comment by Phyroks** [comment]
+1

**Comment by bengivre** [comment]
What is the issue not to merge that ?

**Comment by slumberrad0n** [comment]
What is the current status? When can we expect the function?

**Comment by chakflying** [comment]
Improved design according to suggestions, the color dropdown will now detect the custom hex color and display accordingly. It's not a lot of functionality but I think it's complicated enough as it is, with edge cases I did not think of. 

 - no custom color picker
 - no option to delete monitors with a tag
 - does not handle individual values set with each monitorTag

**Comment by chakflying** [comment]
I also realized all the "Saved" messages from the server are not translated when they are displayed in the UI? Might want to add a way to handle this when displaying a toast.

**Comment by louislam** [comment]
> I also realized all the "Saved" messages from the server are not translated when they are displayed in the UI? Might want to add a way to handle this when displaying a toast.

Yes, @karelkryda had opened a pr for server side translation.
https://github.com/louislam/uptime-kuma/pull/2232

**Comment by bengivre** [comment]
We are getting there, thank you <3

**Comment by tminei** [comment]
Awesome!

File: src/components/TagEditDialog.vue [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    },
+
+    mounted() {
+        this.modal = new Modal(this.$refs.modal);
+    },
+
+    methods: {
+        /**
+         * Show confirmation for deleting a tag
+         * @returns {void}
```
Comment (Computroniks) [inline src/components/TagEditDialog.vue:195]: If the function doesn't return anything I don't think the return line should be included 
```suggestion
```

**Review by Computroniks — CHANGES_REQUESTED** [review]

**Comment by louislam** [comment]
Thanks, work as expected.

**Pull request merged**